staging/src/k8s.io/apiserver/pkg/reconcilers/lease.go

```diff
@@ -94,17 +101,28 @@ func (s *storageLeases) ListLeases() ([]string, error) {
 
 // UpdateLease resets the TTL on a master IP in storage
 // UpdateLease will create a new key if it doesn't exist.
-func (s *storageLeases) UpdateLease(ip string) error {
+func (s *storageLeases) UpdateLease(ip string, apiserverId string, endpointPorts []corev1.EndpointPort) error {
 	key := path.Join(s.baseKey, ip)
+	// TODO: how to use endpointPorts to save in the key
+	if len(endpointPorts) > 0 {
+		key = path.Join(key, fmt.Sprint(endpointPorts[0].Port))
+	}
 	return s.storage.GuaranteedUpdate(apirequest.NewDefaultContext(), key, &corev1.Endpoints{}, true, nil, func(input kruntime.Object, respMeta storage.ResponseMeta) (kruntime.Object, *uint64, error) {
 		// just make sure we've got the right IP set, and then refresh the TTL
 		existing := input.(*corev1.Endpoints)
 		existing.Subsets = []corev1.EndpointSubset{
 			{
 				Addresses: []corev1.EndpointAddress{{IP: ip}},
+				Ports:     endpointPorts,
 			},
 		}
 
+		// update apiserverId
+		if existing.Labels == nil {
+			existing.Labels = map[string]string{}
+		}
+		existing.Labels[APIServerIdentityLabel] = apiserverId
```

Comment:
Is setting this label feature gated indirectly somehow, or will this always happen even if the feature gate is off?  Should it?

If there a test verifying that the right thing happens here when the feature gate is off?

Reply:
Added a condition block to check for the feature gate before doing this update.